qiskit_experiments/framework/composite/parallel_experiment.py

```diff
@@ -89,14 +89,23 @@ def _combined_circuits(self, device_layout: bool) -> List[QuantumCircuit]:
             # Num qubits will be computed from sub experiments
             num_qubits = len(self.physical_qubits)
         else:
-            # Work around for backend coupling map circuit inflation
+            # Expand the number of qubits similar to how qiskit.transpile does
+            # Here we progress from most to least specific way of specifying
+            # the number of qubits: coupling_map->target->backend
+            #
+            # TODO: Behave more like a layout transpiler pass and set the
+            # _layout property on the circuits. Doing this requires accessing
+            # private attributes of Qiskit or possibly running a layout pass of
+            # the transpiler if that can be done without too much overhead.
+            num_qubits = 1 + max(self.physical_qubits)
             coupling_map = getattr(self.transpile_options, "coupling_map", None)
-            if coupling_map is None and self.backend:
-                coupling_map = self._backend_data.coupling_map
+            target = getattr(self.transpile_options, "target", None)
             if coupling_map is not None:
-                num_qubits = 1 + max(*self.physical_qubits, np.max(coupling_map))
-            else:
-                num_qubits = 1 + max(self.physical_qubits)
+                num_qubits = max(num_qubits, 1 + np.max(coupling_map))
+            elif target is not None:
+                num_qubits = max(num_qubits, target.num_qubits)
+            elif self.backend:
```

Comment:
I think we could remove V1 support soon. In this PR, I just wanted to address deprecations in Qiskit 1.3. Removing support would be a bigger, breaking change. At the moment, there is still at least one third party backend provider using BackendV1 but work is in progress to migrate it to BackendV2.

By the way, the main reason I changed this code is because I modified `test_t1_parallel_exp_transpile` to use a `target` instead of a `coupling_map` as a transpile option and the test started failing because the code here was not expanding the circuit based on the size of the target. For completeness, I added expansion based on backend size as well, though generally backends should have coupling maps that have a size that matches `num_qubits`.